content/en/blog/_posts/2025-11-XX-handling-undecryptable-resources-alpha/index.md

```diff
@@ -0,0 +1,228 @@
+---
+layout: blog
+title: "Kubernetes v1.35: Handling Undecryptable Resources (Alpha)"
+date: 2025-11-XXT09:00:00-08:00
+draft: true
+slug: kubernetes-v1-35-handling-undecryptable-resources-alpha
+author: >
+  [Ben Petersen](https://github.com/benjaminapetersen) (Microsoft)
+---
+
+## Handling undecryptable resources: when one broken Secret ruins everything
+
+If you’ve ever rotated an encryption key a little too aggressively, miswired a
+KMS plugin, or inherited a cluster with a mysterious etcd history, you may have
+seen the most frustrating flavour of "encryption at rest": the part where *one
+malformed object* makes it impossible to list **any** objects of that type. A
+single Secret or ConfigMap that can’t be decrypted turns simple operations like
+`kubectl get secrets -n production` into a full-cluster faceplant. And because
+the API server can’t read the object, you can’t delete it either—unless you’re
+willing to go spelunking directly in etcd with all the associated "please don’t
+brick the cluster" anxiety.
+
+KEP-3926, *Handling undecryptable resources*, gives administrators a safer,
+API-native way to (a) find out which resources are failing to decrypt and (b)
+remove them when you really, truly mean it.
+
+## The problem: one bad object, many sad APIs
+
+Encryption at rest in Kubernetes works by transforming objects as they’re
+written to and read from storage. If that transformation fails on the way *out*
+of etcd—because a key is gone, the ciphertext is corrupted, or some other
+storage-layer issue—then:
+
+- Any request that needs to read that object fails.
+- A `LIST` or `WATCH` that touches the same prefix also fails, even if all
+	other objects are perfectly fine.
+- Because the API server can’t decode the object, you can’t just `kubectl
```

Comment:
Not sure how proficient / who the target audience is, but it might be useful to mention that the kube-apiserver tries to read every object, before deleting it and that this part fails horribly and makes deleting of corrupt objects impossible.